graylog2-server/src/main/java/org/graylog2/migrations/V20170607164210_MigrateReopenedIndicesToAliases.java

```diff
@@ -0,0 +1,141 @@
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.graylog2.indexer.gson.GsonUtils.asJsonObject;
+import static org.graylog2.indexer.gson.GsonUtils.asString;
+
+public class V20170607164210_MigrateReopenedIndicesToAliases extends Migration {
+    private static final String REOPENED_INDEX_SETTING = "graylog2_reopened";
+
+    private final Version elasticsearchVersion;
+    private final IndexSetService indexSetService;
+    private final MongoIndexSet.Factory mongoIndexSetFactory;
+    private final Indices indices;
+    private final JestClient jestClient;
+
+    @Inject
+    public V20170607164210_MigrateReopenedIndicesToAliases(Node node,
+                                                           IndexSetService indexSetService,
+                                                           MongoIndexSet.Factory mongoIndexSetFactory,
+                                                           Indices indices,
+                                                           JestClient jestClient) {
+        this.elasticsearchVersion = node.getVersion().orElseThrow(() -> new ElasticsearchException("Unable to retrieve Elasticsearch version."));
+        this.indexSetService = indexSetService;
+        this.mongoIndexSetFactory = mongoIndexSetFactory;
+        this.indices = indices;
+        this.jestClient = jestClient;
+    }
+
+    @Override
+    public ZonedDateTime createdAt() {
+        return ZonedDateTime.parse("2017-06-07T16:42:10Z");
+    }
+
+    // Create aliases for legacy reopened indices.
+    @Override
+    public void upgrade() {
+        this.indexSetService.findAll()
+            .stream()
+            .map(mongoIndexSetFactory::create)
+            .flatMap(indexSet -> getReopenedIndices(indexSet).stream())
+            .forEach(indices::markIndexReopened);
+    }
+
+    private Set<String> getReopenedIndices(final Collection<String> indices) {
+        final String indexList = String.join(",", indices);
+        final State request = new State.Builder().withMetadata().indices(indexList).build();
+
+        final JestResult jestResult = JestUtils.execute(jestClient, request, () -> "Couldn't read cluster state for reopened indices " + indices);
+        final JsonObject indicesJson = getClusterStateIndicesMetadata(jestResult.getJsonObject());
+        final ImmutableSet.Builder<String> reopenedIndices = ImmutableSet.builder();
+
+        for (Map.Entry<String, JsonElement> entry : indicesJson.entrySet()) {
+            final String indexName = entry.getKey();
+            final JsonElement value = entry.getValue();
+            if (value.isJsonObject()) {
+                final JsonObject indexSettingsJson = value.getAsJsonObject();
+                final JsonObject indexSettings = getIndexSettings(indexSettingsJson, indexName);
```

Comment:
Could this be replaced with a JsonPath expression instead of nested function calls?
Essentially we are trying to figure out whether one of  `settings.index.graylog2_reopened` or `archived.settings.index.graylog2_reopened` exists or not for each index we looked at.